src/common/utils.lua

```diff
@@ -333,4 +333,42 @@ function utils.affiliationsForAddress(address, ants)
 	return affiliations
 end
 
+--- @param versionString string
+function utils.validateSemver(versionString)
+	assert(type(versionString) == "string", "Version must be a string")
+
+	-- Reject versions with build metadata (anything after +)
+	if versionString:find("%+") then
+		error("Build metadata is not supported: " .. versionString)
+	end
+
+	-- The pattern now ensures there are exactly 3 version segments
+	-- and optionally a prerelease identifier
+	local pattern = "^(%d+)%.(%d+)%.(%d+)$"
+	local prereleasePattern = "^(%d+)%.(%d+)%.(%d+)%-([%w%.%-]+)$"
+
+	local hasMatch = versionString:match(pattern) or versionString:match(prereleasePattern)
+	if not hasMatch then
+		error("Invalid semver string: " .. versionString)
+	end
+
+	-- If we have a prerelease version, validate its format
+	local prerelease = versionString:match("^%d+%.%d+%.%d+%-([%w%.%-]+)$")
```

Comment:
Is this leftover from an earlier iteration? Seems like you're already testing this above.

Reply:
Will remove - stepping away from semver to iterative versions